repo2docker/buildpacks/repo2docker-entrypoint

```diff
@@ -2,11 +2,19 @@
 # lightest possible entrypoint that ensures that
 # we use a login shell to get a fully configured shell environment
 # (e.g. sourcing /etc/profile.d, ~/.bashrc, and friends)
+
+# Setup a file descriptor (FD) that is connected to a tee process which
+# writes its input to $REPO_DIR/.jupyter-server-log.txt
+# We later use this FD as a place to redirect the output of the actual
+# command to. We can't add `tee` to the command directly as that will prevent
+# the container from exiting when `docker stop` is run.
+exec {log_fd}> >(exec tee $REPO_DIR/.jupyter-server-log.txt)
```

Comment:
This is a very mysterious line! I have no idea what the syntax of `{name}>` means! But clearly it results in the desired effect - `$log_fd` is defined as an integer, e.g. 10. Where did you learn this?

Reply:
The all knowing SO: https://stackoverflow.com/a/55678435